app/services/medium_providers/provider.rb

```diff
@@ -0,0 +1,39 @@
+module MediumProviders
+  class Provider
+    def self.friendly_name
+      self.name
+    end
+
+    def initialize(entity)
+      @entity = entity
+    end
+
+    def medium_uri
+      throw "medium_uri is not implemented for #{self.class.name}"
+    end
+
+    def boot_files_uri
+      throw "boot_files_uri is not implemented for #{self.class.name}"
+    end
+
+    def unique_id
+      @unique_id ||= Base64.urlsafe_encode64(Digest::SHA1.digest(medium_uri.to_s), padding: false)
```

Comment:
I think there is way enough entropy for couple of installation media :-) I suggest to shorten the ID to 6 characters, that's still 2^32 possibilities. Chances of collisions are pretty low, if we put media name prefix that's almost zero.

Reply:
I don't have affinity towards any solution. Personally I don't like taking parts of pseudo-random values, as randomness is not always distributed evenly inside the value. I know it's the case with guids, but I don't have enough knowledge of SHA1.